packages/sdk/src/types/VersionedModuleSource.ts

```diff
@@ -109,33 +108,61 @@ export async function parseModuleInterface(
  * Extract the embedded smart contract schema bytes. Returns `null` if no schema is embedded.
  * @param {VersionedModuleSource} moduleSource The smart contract module source.
  * @returns {RawModuleSchema | null} The raw module schema if found.
+ * @throws If the module source cannot be parsed or contains duplicate schema sections.
  */
-export async function getEmbeddedModuleSchema(
-    moduleSource: VersionedModuleSource
-): Promise<RawModuleSchema | null> {
-    const wasmModule = await WebAssembly.compile(moduleSource.source);
-    const versionedSchema = schemaBytesFromWasmModule(
-        wasmModule,
-        'concordium-schema'
+export function getEmbeddedModuleSchema({
+    source,
+    version,
+}: VersionedModuleSource): RawModuleSchema | null {
+    const sections = findCustomSections(
+        new WebAssembly.Module(source),
+        version
     );
-    if (versionedSchema !== null) {
-        return { type: 'versioned', buffer: versionedSchema };
+    if (!sections) {
```

Comment:
It would be nice to spell this out explicitly, i.e. `!== undefined ...`

Reply:
done